### PR TITLE
SONARJAVA-3658 Fix TextBlocks in SE engine, fix issue location between Text Blocks

### PR DESCRIPTION
**Pull request by margarita-nedzelska-sonarsource**

### CODE REVIEW

**Comment by margarita-nedzelska-sonarsource** [comment]
After testing on different input, I've found out the Text_Blocks are not fully supported. So I ptovided a small fix for SE and for Locations reporting between 2 trees, when the last one is Text Block (it requires to calculate the end of the tree)

File: java-frontend/src/main/java/org/sonar/java/AnalyzerMessage.java [hunk 1]
```diff
@@ -139,29 +142,48 @@ public boolean isEmpty() {
       lastSyntaxToken.line(),
       lastSyntaxToken.column() + lastSyntaxToken.text().length()
     );
+    checkLocation(firstSyntaxToken, location);
+    return location;
+  }
+
+  private static AnalyzerMessage.TextSpan textSpanEndsWithTextBlock(SyntaxToken firstSyntaxToken, SyntaxToken lastSyntaxToken) {
```
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/AnalyzerMessage.java:149]: what about calling this method `textSpanBetween(SyntaxToken firstSyntaxToken, LiteralTree textBlockTree)` to cover this case explicitly instead?

File: java-checks-test-sources/src/main/files/non-compiling/checks/SimpleStringLiteralForSingleLineStringsCheck.java [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 public class SimpleStringLiteralForSingleLineStringsCheck {
   
-  public void str() { // Noncompliant@+1 [sc=23;endColumn=42]]{{Use simple literal for a single-line string.}}
+  public void str() { // Noncompliant@+1 [[sc=23;endColumn=43]]{{Use simple literal for a single-line string.}}
```
Comment (m-g-sonar) [inline java-checks-test-sources/src/main/files/non-compiling/checks/SimpleStringLiteralForSingleLineStringsCheck.java:5]: wow, nice finding and undetected issue in precise location testing... 👍🏻 
worth a ticket to improve the verifier, don't you think?

Comment (margarita-nedzelska-sonarsource) [reply java-checks-test-sources/src/main/files/non-compiling/checks/SimpleStringLiteralForSingleLineStringsCheck.java:5]: Was surprised too. Yep, we should definitely do sth with verify to detect such typos)

**Comment by m-g-sonar** [comment]
LGTM!

**Pull request merged**